src/routes/components/modal/+page.md

```diff
@@ -0,0 +1,75 @@
+<script lang="ts">
+    import Modal from "$lib/components/Modal.svelte";
+    import DocsLoremIpsum from "$docs/components/DocsLoremIpsum.svelte";
+
+    let visible = false;
+    let role: "dialog" | "alert" = "dialog";
+</script>
+
+# Modal
+
+A Modal is a dialog that appears on top of the app's content, and must be dismissed by the app before interaction can resume. It is useful as a select component when there are a lot of options to choose from, or when filtering items in a list, as well as many other use cases.
+
+```html
+<script lang="ts">
+    let visible = false;
+</script>
+
+<button on:click={() => (visible = true)}>
+  Open modal
+</button>
+
+<Modal {visible} on:nnsClose={() => (visible = false)}>
+  <svelte:fragment slot="title">My title</svelte:fragment>
+
+  <p>My content</p>
+</Modal>
+```
+
+## Properties
+
+| Property               | Description                                                                                             | Type                    | Default     |
+| ---------------------- | ------------------------------------------------------------------------------------------------------- | ----------------------- | ----------- |
+| `visible`              | Display or hide the modal.                                                                              | `boolean`               | `false`     |
+| `role`                 | The modal is either a dialog meant for getting work done or an alert that requires immediate attention. | `dialog` or `alert`     | `dialog`    |
+| `testId`               | Add a `data-tid` attribute to the DOM, useful for test purpose.                                         | `string` or `undefined` | `undefined` |
+| `disablePointerEvents` | Disable interactive elements - close actions - of the modal.                                            | `boolean`               | `false`     |
+
+## Slots
+
+| Slot name    | Description                                                                                  |
+| ------------ | -------------------------------------------------------------------------------------------- |
+| Default slot | The content of the modal.                                                                    |
+| Title        | The title of the modal. Displayed in a toolbar with a "Close" icon button on the right side. |
```

Comment:
What do you think about using the lowercase version `title` to reflect the exact slot value?

Reply:
Absolutely, that's a typo 😅

Reply:
Done.